xarray/backends/rasterio_.py

```diff
@@ -5,127 +5,109 @@
 except ImportError:
     rasterio = False
 
-from .. import Variable
-from ..core.utils import FrozenOrderedDict, Frozen, NDArrayMixin, is_scalar
+from .. import DataArray
+from ..core.utils import NDArrayMixin, is_scalar
 from ..core import indexing
-from ..core.pycompat import OrderedDict, suppress
 
-from .common import AbstractDataStore
-
-_rio_varname = 'raster'
-
-_error_mess = ('The kind of indexing operation you are trying to do is not '
-               'valid on RasterIO files. Try to load your data with ds.load()'
-               'first.')
+_ERROR_MSG = ('The kind of indexing operation you are trying to do is not '
+              'valid on rasterio files. Try to load your data with ds.load()'
+              'first.')
 
 
 class RasterioArrayWrapper(NDArrayMixin):
-    def __init__(self, ds):
-        self.ds = ds
-        self._shape = self.ds.count, self.ds.height, self.ds.width
+    """A wrapper around rasterio dataset objects"""
+    def __init__(self, riods):
+        self.riods = riods
+        self._shape = self.riods.count, self.riods.height, self.riods.width
         self._ndims = len(self.shape)
 
     @property
     def dtype(self):
-        return np.dtype(self.ds.dtypes[0])
+        return np.dtype(self.riods.dtypes[0])
```

Comment:
This is a good question. I just assume it *has* to be so (since rasterio's ``get_data`` returns a ndarray of uniform dtype) but I can't be sure. I added a check for this now.

If the bands have different dtypes they should probably be handled as different variables, which would be bad for the current dataarray design.

@perrygeo @sgillies if you have time could you comment here?